src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -2707,6 +2707,12 @@ void Unit::_UpdateSpells(uint32 time)
         }
     }
 
+    for (auto nexctSpellCrit : m_spellCritMap)
+    {
+        if (!nexctSpellCrit.first.first)
```

Comment:
could you explain more in detail what this check does and how it does it please ?